src/webapi/request_animation_frame_handle.rs

```diff
@@ -0,0 +1,23 @@
+/// Struct representing an animation frame request
+/// Use webapi::window::request_animation_frame to create this
+#[derive(Debug)]
+pub struct RequestAnimationFrameHandle {
+    id: u32,
+}
+
+impl RequestAnimationFrameHandle {
+    /// This id comes from here:
+    /// [(Javascript docs)](https://developer.mozilla.org/en-US/docs/Web/API/window/requestAnimationFrame)
+    pub fn new(id: u32) -> RequestAnimationFrameHandle {
+        RequestAnimationFrameHandle {
+            id,
+        }
+    }
+    /// This cancels an animation frame request
+    /// [(Javascript docs)](https://developer.mozilla.org/en-US/docs/Web/API/Window/cancelAnimationFrame)
+    pub fn cancel(&self) {
+        js!{
+            cancelAnimationFrame(@{self.id});
```

Comment:
This is not enough; if someone cancelled this *before* it was triggered then the callback needs to be dropped too, or it will leak memory.

Reply:
In order to manage this, I had to do some stuff that might be a bit weird. I basically stored the whole handle in javascript land and just used it in inlined javascript.